### ISSUE TITLE
Feature request: add support for directives

### ISSUE THREAD

**Issue by abedeoan** [graphql-java-generator/graphql-maven-plugin-project#4]
Doesn't the graphql-maven-plugin support directives (e.g. https://www.apollographql.com/docs/graphql-tools/schema-directives/) or there is something wrong with the below directive definition?

`directive @defer (reason: String = "some reason why") on FIELD`

The error message received is:
`[ERROR] Failed to execute goal com.graphql-java-generator:graphql-maven-plugin:1.1-SNAPSHOT:graphql (default) on project inventory-generatedmodel: Unknown node type: graphql.language.DirectiveDefinition DirectiveDefinition{name='defer', inputValueDefinitions=[InputValueDefinition{name='reason', type=TypeName{name='String'}, defaultValue=StringValue{value='some reason why'}, directives=[]}], directiveLocations=[DirectiveLocation{name='FIELD'}]} -> [Help 1]`
while generating the code in "client" mode.

Any recommendation would be appreciated.

**Comment by etienne-sf** [comment]
DirectiveDefinition  are currently not managed. 
It's on the TODO list...

**Comment by etienne-sf** [comment]
I'll make the plugin more tolerant with GraphQL features that it doesn't manage

**Comment by abedeoan** [comment]
I can confirm that, with version 1.3.1, the error from the initial post does not occur anymore.

**Comment by etienne-sf** [comment]
Right. It is not really supported.

I'm working on it.

**Comment by abedeoan** [comment]
Thank you for the update!

**Comment by etienne-sf** [comment]
I released the 1.5.0 yesterday. 

Can you confirm that this issue is solved ?

**Comment by DirkMahler** [comment]
Just tried but ran into a problem: we're using the Neo4j GraphQL Java library (https://github.com/neo4j-graphql/neo4j-graphql-java) on the server side and from that directives to specify relations:

```
type Person {
  name : String
  actedIn: [Movie] @relation(name:"ACTED_IN", direction:OUT)
}
```

This results in an error during code generation:

```
The directive named 'relation' could not be found
```

If we add the directive definition...

```
directive @relation(
    name: String
    direction: direction
) on FIELD_DEFINITION

enum direction {
    IN
    OUT
}
```

...to our schema another problem occurs:

```
Caused by: java.lang.RuntimeException: Error when trying to execute 'getValue' on 'graphql.language.EnumValue': graphql.language.EnumValue.getValue()
    at com.graphql_java_generator.GraphqlUtils.invokeMethod (GraphqlUtils.java:366)
    at com.graphql_java_generator.plugin.DocumentParser.readAppliedDirectives (DocumentParser.java:477)
    at com.graphql_java_generator.plugin.DocumentParser.readFieldTypeDefinition (DocumentParser.java:719)
    at com.graphql_java_generator.plugin.DocumentParser.readField (DocumentParser.java:694)
    at com.graphql_java_generator.plugin.DocumentParser.lambda$readObjectType$12 (DocumentParser.java:534)
```

**Comment by etienne-sf** [comment]
Hum, hum,

  So you're using the client part of this plugin, right ?
Which version ?

Can you provide the GraphQL schema you use, so that we can repeat this error ?

**Comment by etienne-sf** [comment]
And by the way, please don't pollute existing threads. 
Please create your own, as this is a specific issue.

**Comment by DirkMahler** [comment]
Created a separate issue: https://github.com/graphql-java-generator/graphql-maven-plugin-project/issues/35

**Comment by etienne-sf** [comment]
Thank you
:)

**Comment by etienne-sf** [comment]
Support for the directive has been released in the 1.5.0 version
